lib/resque/data_store.rb

```diff
@@ -91,10 +92,16 @@ def all_resque_keys
     end
 
     def server_time
-      time, _ = @redis.time
+      time, _ = @redis_time_available ? @redis.time : Time.now
       Time.at(time)
     end
 
+    def redis_time_available?
+      @redis.time
+    rescue Redis::CommandError
```

Comment:
Does this error come from the server? i.e. do we need to send this command to the server in order to know that it's not supported? should we cache the response so that we don't have to ping the server every time?